src/App.tsx

```diff
@@ -11,74 +17,74 @@ import EnterSpaceIdPage from "./pages/EnterSpaceIdPage";
 
 import { ProtectedRoute } from "./components/ProtectedRoute";
 
+const RedirectToSelectAction = () => {
+  const { spaceId } = useParams();
+  return <Navigate to={`/space/${spaceId}/select-action`} />;
+};
+
```

Comment:
The RedirectToSelectAction component is redundant in this case. You can directly use Navigate in the Route definition:
```js
<Route 
  path="/space/:spaceId" 
  element={<Navigate to={`/space/:spaceId/select-action`} replace />} 
/>
```